src/main/java/com/example/demo/repository/book/spec/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,22 @@
+package com.example.demo.repository.book.spec;
+
+import com.example.demo.model.Book;
+import com.example.demo.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String authorPart) {
+        return (root, query, criteriaBuilder) ->
+                criteriaBuilder.like(criteriaBuilder
```

Comment:
if I remember the correct operator LIKE ignore case in MySQL

Reply:
following SOLID principle, to provide durability and exclude some register sensitive cases in future, I' will leave this transfering